src/core.js

```diff
@@ -1,7 +1,7 @@
 
 var oldInit = jQuery.fn.init,
 	oldIsNumeric = jQuery.isNumeric,
-	rattrHash = /\[\s*\w+\s*[~|^$*]?=\s*(?![\s'"])[^#\]]*#/;
+	rattrHash = /\[(\s*[-\w]+\s*)([~|^$*]?=)\s*([-\w#]+)\s*\]/g;
```

Comment:
This change opened up matches that don't include `#` characters. I think the last capture group should be `([-\w#]*?#[-\w#]*)`.